src/main/java/com/sda/eventapp/service/EventService.java

```diff
@@ -36,29 +36,32 @@ public class EventService {
     private final ImageService imageService;
     private final EventMapper mapper;
 
-    public Event save(CreateEventForm form, User owner, MultipartFile file) {
+    public Event save(EventForm form, User owner, MultipartFile file) {
         form.setOwner(owner);
         form.setImage(solveImage(file));
         return repository.save(mapper.toEvent(form));
     }
 
-    public Event update(CreateEventForm form) {
-        Event event = repository.findById(form.getId())
-                .orElseThrow(() -> new RuntimeException("Event not found"));
-        event.setId((form.getId()));
-        event.setTitle((form.getTitle()));
-        event.setDescription(form.getDescription());
-        event.setStartingDateTime(form.getStartingDateTime());
-        event.setEndingDateTime(form.getEndingDateTime());
-
-        return repository.save(event);
+    public Event update(EventForm form, MultipartFile file) {
```

Comment:
I've rearranged it a little bit:

```
public Event update(EventForm form, MultipartFile file) {
        Event event = this.findById(form.getId());

        form.setOwner(event.getOwner());

        if (file.getOriginalFilename() != null && !file.getOriginalFilename().isBlank()) {
            // if new file was uploaded
            form.setImage(saveImageLocally(file));
        } else {
            // otherwise get image from db
            form.setImage(event.getImage());
        }

        return repository.save(mapper.toEvent(form));
    }
```